prow/pod-utils/decorate/podspec.go

```diff
@@ -286,7 +288,13 @@ func decorate(spec *kube.PodSpec, pj *kube.ProwJob, rawEnv map[string]string) er
 		}),
 		VolumeMounts: []kube.VolumeMount{logMount, gcsCredentialsMount},
 	})
-	spec.Volumes = append(spec.Volumes, append(sshKeysVolumes, logVolume, codeVolume, toolsVolume, gcsCredentialsVolume)...)
+	spec.Volumes = append(spec.Volumes, logVolume, toolsVolume, gcsCredentialsVolume)
+
+	if willCloneRefs {
```

Comment:
nit: This code would be a little simpler to read if there was just one `if willCloneRefs {}` block instead of three. The contents of this conditional block could be moved to the end of the first `if willCloneRefs {}` block.  The other `if willCloneRefs {}` block could also be condensed into the first if the init container options are defined earlier.

Reply:
Yeah I debated that, but it was going to break up the "set up vars, build the container, add to the pod" pattern that we had before. I'll look at bunching it together but it may make more sense to keep these things grouped by (init)container

Reply:
I made the changes but ultimately I think it's cleaner as-is with three blocks. Otherwise you need to jump all around the code to determine how the `initupload` container is set up and the ordering of the logic here gets all weird.